pepper-apis/src/main/java/org/broadinstitute/ddp/service/actvityinstancebuilder/form/FormInstanceCreatorHelper.java

```diff
@@ -86,6 +89,18 @@ public int setDisplayNumbers(FormInstance formInstance) {
         return startingNumber;
     }
 
+    /**
+     * If snapshotted address GUID is preserved in activity instance substitutions (with key ADDRESS_GUID)
+     * then find MAILING_ADDRESS component in the created formInstance and set addressGuid with this found
+     * snapshotted address GUID
+     */
+    public void populateSnapshottedAddress(MailingAddressComponent mailingAddressComponent, Map<String, String> activitySnapshots) {
+        String addressGuid = activitySnapshots.get(I18nTemplateConstants.Snapshot.ADDRESS_GUID);
+        if (addressGuid != null) {
```

Comment:
Should we check mailing-address component is not null?

Reply:
Sure, thanks!
Fixed.